src/Symfony/Component/HttpFoundation/Tests/BinaryFileResponseTest.php

```diff
@@ -314,7 +314,15 @@ public function testXAccelMapping($realpath, $mapping, $virtual)
         $property->setValue($response, $file);
 
         $response->prepare($request);
-        $this->assertEquals($virtual, $response->headers->get('X-Accel-Redirect'));
+        $header = $response->headers->get('X-Accel-Redirect');
+
+        if ($header) {
+            // Making sure the path doesn't contain characters unsupported by nginx
+            $this->assertDoesNotMatchRegularExpression('/%(?![0-9A-F]{2})|\?/', $header);
```

Comment:
Not 100% sure, but with 
```suggestion
            $this->assertMatchesRegularExpression('/^([^?%]|%[0-9A-F]{2})*$/', $header);
```
we would reduce the complexity for the reader, as we don't have a negative check with a negative lookahead, right?

Reply:
I tested it, it works as expected